### PR TITLE
Chap13

### PR DESCRIPTION
**Pull request by suuhv**

### CODE REVIEW

**Comment by suuhv** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: [:create, :destroy]
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build micropost_params
```
Comment (tuansaker1412) [inline app/controllers/microposts_controller.rb]: them mot dong sau dong nay

File: app/models/user.rb [hunk 2]
```diff
@@ -49,6 +51,33 @@ def current_user? user
...
+  def send_password_reset_email
+    UserMailer.password_reset(self).deliver_now
+  end
+
+  def password_reset_expired?
+    reset_sent_at < Settings.user.time_save.hours.ago
+  end
+
+  def feed
+    Micropost.where("user_id = ?", id)
```
Comment (tuansaker1412) [inline app/models/user.rb]: remove ()

File: app/uploaders/picture_uploader.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class PictureUploader < CarrierWave::Uploader::Base
+
+  include CarrierWave::MiniMagick
+  process resize_to_limit: [400, 400]
```
Comment (tuansaker1412) [inline app/uploaders/picture_uploader.rb]: Settings

File: app/views/shared/_micropost_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<%= form_for(@micropost) do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
```
Comment (tuansaker1412) [inline app/views/shared/_micropost_form.html.erb]: nhay doi "

File: db/seeds.rb [hunk 5]
```diff
@@ -17,3 +17,9 @@
     activated: true,
     activated_at: Time.zone.now
 end
+
+users = User.order(:created_at).take(6)
```
Comment (tuansaker1412) [inline db/seeds.rb]: remove ()

File: db/seeds.rb [hunk 6]
```diff
@@ -17,3 +17,9 @@
...
+Settings.micropost.size.times do
+  content = Faker::Lorem.sentence(5)
```
Comment (tuansaker1412) [inline db/seeds.rb]: remove ()

File: app/assets/stylesheets/custom.scss [hunk 7]
```diff
@@ -194,3 +194,49 @@ input {
...
+    }
+  }
+  .timestamp {
+    color: $gray-light;
+    display: block;
+    margin-left: 60px;
+  }
+  .gravatar {
+    float: left;
+    margin-right: 10px;
```
Comment (tuansaker1412) [inline app/assets/stylesheets/custom.scss]: gop dong nay vs dong duoi thanh mot

File: app/models/micropost.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+
+  default_scope -> {order(created_at: :desc)}
```
Comment (tuansaker1412) [inline app/models/micropost.rb]: indent ->{ viet lien
khong dung default_scope, scope tu the

**Comment by huongnguyenmta** [comment]
conflict

**Comment by huongnguyenmta** [comment]
conflicted

**Comment by huongnguyenmta** [comment]
commented

**Pull request closed without merging**